KevinHardy-Cooper/VApp
# Add ESLint and Make Fixes for V&V

# Description
Code linting is a type of static code analysis. It frequently used to find problematic patterns or code that doesn't adhere to certain style guidelines, highlighting error such as stylistic issues, bugs, possible memory leaks, etc. Linting is highly used for web applications in the industry and well supported by the open source community.

ESLint is a linting tool for identifying and reporting on patterns found in ECMAScript/JavaScript code. This free-to-use open source tool will be used to automatically perform static analysis for the application.

Each developer will install the ESLint dependency in their development environment so that their code is automatically analyzed and fixed before submitting pull requests. Additionally, each developer will be using the same custom rule set, which uses recommended settings by ESLint as well as some custom configuration.

# NOTE
The files changed in this PR are from running `eslint --fix src\`. All of the files should now be consistently styled. The only new file is the `.eslintrc.json` file which defines the rule set used to make all the files consistently styled.

# How to Test
- Checkout the branch with `git checkout TERIN_add_static_analysis`
- Install ESLint with `npm install -g eslint`
- Run the linter with `eslint src\` (NOTE: might be a forward slash on Windows)
- Make sure it runs with no errors, ESLint doesn't provide a success message
- Make sure the application stills runs as normal
